### ISSUE TITLE
Release 15.4.0

### ISSUE THREAD

**Issue by mattxwang** [stylelint/stylelint#6743]
- [x] stylelint release
- [x] stylelint-config-recommended update/release
- [x] stylelint-config-standard update/release
- [x] stylelint.io update
- [x] stylelint-demo check
- [ ] tweet

Following up from the conversation in #6725. To summarize, we have a new rule, new CLI flag + new alias, one rule bugfix, and one performance fix. See the [in-progress changelog](https://github.com/stylelint/stylelint/pull/6725/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed).

We could wait on #6742 depending on its complexity; curious for other thoughts!

(I've never done a release before, though I'm happy to take it up if others don't have time. I may not have some permissions though!)

**Comment by ybiquitous** [comment]
@mattxwang Thanks for opening this issue. I think it's OK to release 15.4.0 (maybe this weekend).

> I may not have some permissions though!

Unfortunately, you don't have permission to publish packages to the npm registry, and I cannot grant you permission due to my lack of rights. 😓 
(cc @jeddy3)

So, I'll take over at this time. I appreciate your proposal. 😄

**Comment by ybiquitous** [comment]
> We could wait on #6742 depending on its complexity; curious for other thoughts!

Merged. 👍🏼 

So, we can ship our standard config together:

- https://github.com/stylelint/stylelint-config-standard/issues/275

**Comment by ybiquitous** [comment]
I'd like to include the following in v15.4.0:

- #6747

**Comment by mattxwang** [comment]
Sounds good, just dropped a review.

for https://github.com/stylelint/stylelint-config-standard/issues/275, would you like me to open a PR in the repo to turn it on? Happy to do so, but not sure if it's already in progress.

**Comment by ybiquitous** [comment]
> for [stylelint/stylelint-config-standard#275](https://github.com/stylelint/stylelint-config-standard/issues/275), would you like me to open a PR in the repo to turn it on? Happy to do so, but not sure if it's already in progress.

@mattxwang Thanks for your proposal. Please do so!

**Comment by ybiquitous** [comment]
It might be good timing to publish also `create-stylelint`: https://github.com/stylelint/create-stylelint/compare/0.2.0...main

**Comment by mattxwang** [comment]
No worries at all! Confirming on my end that I've received the invitations.

Would you prefer to continue taking point on this release? Happy to help out however possible!

**Comment by ybiquitous** [comment]
@mattxwang Could you try the release workflow if you are available this weekend? Our [maintainer guide](https://stylelint.io/maintainer-guide/releases) should help you.

I guess you don't have access to our Twitter account, so I'll do only the last job of tweeting the announcement instead of you. (@jeddy3 manages all accounts related to Stylelint, not me)

Feel free to ask me if you have any questions.

**Comment by mattxwang** [comment]
Sounds good, will do!

**Comment by mattxwang** [comment]
Hi @ybiquitous, sorry to bother again - unfortunately, I don't have permissions to force push to `main` (i.e. the same problem as create-stylelint). Would you be able to help me here?

**Comment by ybiquitous** [comment]
@mattxwang No problem. I just changed the repo setting. Can you try it again?

**Comment by mattxwang** [comment]
Hm, no dice @ybiquitous - same error as before (protected branch hook declined). Tried `git push` and `git push --force`.

**Comment by ybiquitous** [comment]
Umm. Can you paste the error log here? You should have push access to the main branch 🤔 

<img width="798" alt="Screenshot 2023-04-01 at 13 09 49" src="https://user-images.githubusercontent.com/473530/229264975-55c3a071-4f54-4c67-acfb-98c85206c70c.png">

**Comment by mattxwang** [comment]
Sure!

```
$ git push
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: At least 1 approving review is required by reviewers with write access. 10 of 10 required status checks are expected.
To github.com:stylelint/stylelint.git
 ! [remote rejected]   main -> main (protected branch hook declined)
error: failed to push some refs to 'github.com:stylelint/stylelint.git'
```

```
$ git push --force
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: At least 1 approving review is required by reviewers with write access. 10 of 10 required status checks are expected.
To github.com:stylelint/stylelint.git
 ! [remote rejected]   main -> main (protected branch hook declined)
error: failed to push some refs to 'github.com:stylelint/stylelint.git'
```

**Comment by ybiquitous** [comment]
Thanks. Is the same error raised when running `npm run release`?

**Comment by mattxwang** [comment]
Hm, so all the steps to publish to `npm` did work - e.g. there is a `15.4.0` of Stylelint. 

When I first ran `npm run release`, everything worked up until pushing the tag; this is where I first got the error.

Now, my `main` has one extra commit:

```
$ git log
commit 5b4a8e3ab2579fc0c4eb7c363e034a568d1be053 (HEAD -> main, tag: 15.4.0)
Author: Matthew Wang <matt@matthewwang.me>
Date:   Fri Mar 31 20:32:56 2023 -0700

    15.4.0
```

If I run `npm run release` again after-the-fact, it prompts be about creating a new version - which I assume is not what we'd want.

Any advice on what to do? Could it be that the branch protection rule is "more important" than the branch permission you sent above? I know repositories have a feature that prevents admins from force-pushing as well.


> (BTW, we should avoid force-push to the main branch 😅 )

Agreed, it felt a bit strange to type this out. I'll stick to `git push` from now on!

**Comment by ybiquitous** [comment]
OK, you are now admin in the repo. Can you try push again?

**Comment by mattxwang** [comment]
Gotcha, that worked - just ran `git push`. Feel free to de-admin me now!

**Comment by mattxwang** [comment]
On a related note - I'm also not able to do the same for stylelint-config-standard (https://github.com/stylelint/stylelint-config-standard/pull/284). Is a temporary admin the solution that works in this situation? Or, would you prefer to explore a different option (since it certainly feels awkward)?

**Comment by ybiquitous** [comment]
I just updated the following repos' settings again to make things simple:
(you are now admin in the repos)

- stylelint
- stylelint-config-recommended
- stylelint-config-standard

**Comment by ybiquitous** [comment]
@mattxwang Sorry for the inconvenience. 🙇🏼 

I will recommend you for an owner-team member when @jeddy3 comes back, since you have a passion for Stylelint!

**Comment by mattxwang** [comment]
> Sorry for the inconvenience.

Haha, not a problem at all - very sympathetic, onboarding new people and permissions is always tricky!

> I will recommend you for an owner-team member ... since you have a passion for Stylelint!

Aw, thank you! Also understand if that doesn't end up being possible for whatever reason (I am still very new in the grand scheme of things); just excited to help out however I can 😊

**Comment by mattxwang** [comment]
After merging https://github.com/stylelint/stylelint.io/pull/311, I've verified that the site has been updated (ex the Changelog has been updated, the text for `declaration-block-no-redundant-longhand-properties` is updated); and, the [demo](https://stylelint.io/demo/) is updated as well; for example, [here is a reproducible demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcCBOEIBcAEwA6A7DGAtIQGarwYCMADgB4DceAvnngIZZ4FoTUUBubJAApiAIyEBKRriYgANCBIBLADYwAcmwC2cRDFo7q6gHRgAzuYXgIuFQHMEIHPgzYQBtDFxRz7igDaXARuIOZoAJ7qqsq4aISQdsr2hOFsPkJQ7sEAuvLB7kgArup+CJyuBO64EIRFuADWNQDuuAlF4RDahNQo1DBIaMowZRRoxTDBLLLWiQ4AYhBI2mxoTgBW5rbWsNRWiC5VYZHRsWj+oaqrI+cKBcdRMDFxCbYOqWjpUJkX7lde4WyMyYQA) for the new `no-unknown-custom-properties` rule!

I think all that's left is now the tweet, unless I've missed anything else. Thanks for your timely support @ybiquitous!
